terraform/modules/jaspersoft/box.tf

```diff
@@ -0,0 +1,62 @@
+resource "aws_security_group" "jaspersoft_server" {
+  vpc_id      = var.vpc_id
+  description = "Jaspersoft server instance"
+}
+
+# tfsec:ignore:aws-vpc-no-public-egress-sgr
+resource "aws_security_group_rule" "jaspersoft_server_egress" {
+  type              = "egress"
+  description       = "Allow all egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = -1
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.jaspersoft_server.id
+}
+
+resource "aws_security_group_rule" "jaspersoft_server_ssh_ingress" {
+  type                     = "ingress"
+  from_port                = 22
+  to_port                  = 22
+  protocol                 = "tcp"
+  source_security_group_id = var.allow_ssh_from_sg_id
+  description              = "SSH from bastion"
+  security_group_id        = aws_security_group.jaspersoft_server.id
+}
+
+resource "aws_security_group_rule" "jaspersoft_server_http_ingress" {
+  type                     = "ingress"
+  from_port                = 8080
+  to_port                  = 8080
+  protocol                 = "tcp"
+  source_security_group_id = aws_security_group.alb.id
+  description              = "HTTP on 8080 from ALB"
+  security_group_id        = aws_security_group.jaspersoft_server.id
+}
+
+resource "aws_instance" "jaspersoft_server" {
+  ami           = "ami-00f499a80f4608e1b" #	eu-west-1 bionic 18.04 LTS amd64
+  instance_type = "t3.medium"
+  tags          = { Name = "${var.prefix}jaspersoft-server" }
+
+  subnet_id                   = var.private_instance_subnet.id
+  vpc_security_group_ids      = [aws_security_group.jaspersoft_server.id]
+  key_name                    = var.ssh_key_name
```

Comment:
Re-using the bastion server's admin SSH key? Seems wrong. No one should need to use the bastion server's admin key.

But people will potentially want to ssh to the jaspersoft server (via the bastion box). I think you could create a dedicated keypair within terraform for this.

I would suggest copying the private key onto the bastion box but maybe not worth it and not simple since we're using the softwire module for it.

Reply:
Set up a separate key for it now, and made it a (sensitive) output